src/main/java/duke/logic/TaskList.java

```diff
@@ -0,0 +1,64 @@
+package duke.logic;
+
+import java.util.ArrayList;
+
+import duke.task.DukeTask;
+
+/**
+ * Represents the list containing all of user's <code>DukeTask</code>.
+ * It contains an <code>ArrayList</code> to track the user's tasks
+ * and methods to operate on it.
+ */
+public class TaskList {
```

Comment:
Perhaps instead of making this ArrayList Object into a class and only exposing the methods that are required, you can consider using a Storage class from which you can accomplish the same goal but have the flexibility of having more methods related to storage rather than restrict yourself to methods only relating to one specific data structure!